### DISCUSSION TITLE
Custom location

### DISCUSSION THREAD

**Discussion by webysther** [shlinkio/shlink#1837] [Help wanted] [answered]
There's a way to create a custom location for the client?

I want to do this configuration on my homelab without the need to burn another domain:

https://url.example.com -> backend/api
https://url.example.com/admin -> client

I tried using proxy pass but the client don't seems not respect the current custom location, I added the assets of client too but no luck...

**Comment by acelaya** [comment]
I would suggest using a subdomain, because it's just easier to manage.

https://url.example.com -> backend/api
https://admin.example.com -> client

That said, if there's still some reason for you to use a sub path, this might be helpful https://github.com/shlinkio/shlink-web-client/discussions/793. It's not exactly the same, but could give you some ideas.

**Comment by webysther** [comment, marked as answer]
I got this work with this solution:

```nginx
location / {
  proxy_pass http://shlink-api:8080/;
}

location /admin/ {
  proxy_pass http://shlink:80/;
  sub_filter_types "*";
  sub_filter_once off;
  # html
  sub_filter 'href="/' 'href="/admin/';
  sub_filter 'content="/' 'content="/admin/';
  sub_filter 'src="/' 'src="/admin/';
  # assets
  sub_filter 'homepage:XY' 'homepage:"/admin"';
  # manifest
  sub_filter '"start_url":"/"' '"start_url":"/admin/"';
  sub_filter '"scope":"/"' '"scope":"/admin/"';
}
```

With this I [don't need to build the client](https://github.com/shlinkio/shlink-web-client#serve-project-in-subpath) again to got work

**Comment by webysther** [comment]
Maybe add to documentation, will be great because with this the security restriction to /s can be done easy.
